ci-operator/step-registry/hypershift/mce/kubevirt/create/disconnected/hypershift-mce-kubevirt-create-disconnected-chain.yaml

```diff
@@ -0,0 +1,207 @@
+chain:
+  as: hypershift-mce-kubevirt-create-disconnected
```

Comment:
I'd like to see less duplication here. The existing [hypershift-mce-kubevirt-create-chain.yaml](https://github.com/openshift/release/blob/master/ci-operator/step-registry/hypershift/mce/kubevirt/create/hypershift-mce-kubevirt-create-chain.yaml) handles most of what this new disconnected chain performs. That existing chain even has an ENABLE_ICSP env var already.


The mirroring of the rhcos image could be added as a separate step or even added directly to the existing chain.

Reply:
Yes, form a functional point of view yes.
On the other side there is a big caveat here:
almost all the script in this case is enclosed within
```
        # shellcheck disable=SC2087
        ssh "${SSHOPTS[@]}" "root@${IP}" bash - << EOF
        set -xeo pipefail
        ...
        EOF
````
because we are executing it remotely over ssh on the bastion host that simply doesn't exist for the connected scenario.
We can eventually try removing the last steps and execute them as part of the test job calling [hypershift-mce-kubevirt-create-chain.yaml](https://github.com/openshift/release/blob/master/ci-operator/step-registry/hypershift/mce/kubevirt/create/hypershift-mce-kubevirt-create-chain.yaml) after the initial preparation done here.